server/lib/processDonation.js

```diff
@@ -0,0 +1,134 @@
+const gateways = require('../gateways');
+const constants = require('../constants/transactions');
+const roles = require('../constants/roles');
+const emailLib = require('./email');
+
+module.exports = (Sequelize, donation) => {
+
+  const services = {
+
+    stripe: (donation) => {
+      const group = donation.Group;
+      const user = donation.User;
+      const paymentMethod = donation.PaymentMethod;
+      const subscription = donation.Subscription;
+      var groupStripeAccount, charge;
+
+      return group.getStripeAccount()
+        .then(stripeAccount => groupStripeAccount = stripeAccount)
+
+        // get or create a customer
+        .then(() => paymentMethod.customerId || gateways.stripe.createCustomer(
+          groupStripeAccount,
+          paymentMethod.token, {
+            email: user.email,
+            group
+          }))
+        .tap(customer => paymentMethod.customerId ? null : paymentMethod.update({customerId: customer.id}))
+
+        // create plan + subscription or a charge for one-time
+        .then(() => {
+          // create a plan + subscription
```

Comment:
those comments are the proof that this should be put aside as a separate function `createPlan`, `createOneTimeCharge`, etc.